src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -218,11 +218,13 @@ public static async Task<PackageSpec> GetProjectSpec(IDependencyGraphProject pro
         {
             var dgSpec = new DependencyGraphSpec();
 
-            var projects = (await solutionManager.GetNuGetProjectsAsync()).OfType<IDependencyGraphProject>();
+            var uniqueProjectDependencies = new HashSet<string>(PathUtility.GetStringComparerBasedOnOS());
 
-            foreach (var project in projects)
+            var projects = ((await solutionManager.GetNuGetProjectsAsync()).OfType<IDependencyGraphProject>()).ToList();
+
+            for (var i=0; i< projects.Count; i++)
```

Comment:
You can simply `foreach` here.

Reply:
`Foreach` on a list still creates a IEnumable instance and waste double memory. This might not make a huge difference but a good practice to use `for` for list instead of `foreach`